src/uPlot.js

```diff
@@ -793,7 +814,10 @@ export default function uPlot(opts, data) {
 				toggleDOM(ip, onOff);
 			}
 
-			resetScale(s.scale);
+			// reset scale if current serie is actually the min or max bound
+			let sc = scales[s.scale]
+			if (!s.minMax || sc.rawMin >= s.minMax[0] || sc.rawMax <= s.minMax[1])
```

Comment:
i don't think `>=` and `<=` are correct. they should probably be strict `>` and `<`, otherwise this will have the same issue as the previous version, that it will reset the scale even if there is another series with the same min or the same max. right?

Reply:
as i mentioned in https://github.com/leeoniya/uPlot/issues/40#issuecomment-549065923, i think we need to make sure that the toggled series is exceeds all other mins or all other maxes of all series shown on the same scale. if a min or max is the same as any another one, it shouldn't invalidate.

Reply:
Actually, the test is right. It's counter intuitive because the `sc.rawMin` is computed with `s.min` so `sc.rawMin > s.min` won't be true (unless the series was hidden and the toggle shows it).
But considering that I won't be storing `rawMin` anymore I'll just recompute it with (or without) the new series bounds. It should be simpler to understand and maintain.

Reply:
try it with this dataset:

```js
[
  [1,2,3,4],
  [1,2,1,2],
  [2,1,2,1],
]
```

toggling one should never invalidate the scale or paths.

Reply:
Thanks for the example. I rethink it 😅.

Reply:
i'm looking through more of this task and realizing this may not have been `good first issue` D:

i'm writing a patch myself to see how many more places need to be touched. so you maybe hold off on any additional work for now. sorry 😭

Reply:
Ok no problem. I tried 🤣, but it was a good way to get in the code and learn somethings (like let doesn't declare a constant 🤓). I'll probably try again

Thanks a lot for your reviews.